### PR TITLE
Linearize script update (hash byte reversal and Python 3 support)

### PR DESCRIPTION
**Pull request by droark**
- Add support for reversed hash bytes.
- Add support for Python 3. The linearization scripts can now run on Python 2 or 3.

Continuing #9304.

### CODE REVIEW

**Comment by droark** [comment]
Found a small bug affecting Python 3. Will update in a bit.

**Comment by droark** [comment]
Nevermind. It wasn't a bug after all. Once I clarified things (command line foolishness on my behalf), I ran Python 2 and 3 with and without the bytes reversed, then ran *shasum -a 256 bootstrap.dat* each time the bootstrap file was written. The results were the same.

While here, I did add a few comments and then did a fixup and rebase.

**Comment by droark** [comment]
Added some more documentation updates but seem to have borked my fixup. Will fix it momentarily.

**Comment by droark** [comment]
Back to normal. Did a quick sanity check. Everything looks good.

File: contrib/linearize/README.md [hunk 1]
```diff
@@ -1,33 +1,47 @@
...
     $ ./linearize-hashes.py linearize.cfg > hashlist.txt
 
 Required configuration file settings for linearize-hashes:
-* RPC: rpcuser, rpcpassword
+* RPC: `rpcuser`, `rpcpassword`
 
 Optional config file setting for linearize-hashes:
-* RPC: host, port
-* Block chain: min_height, max_height
+* RPC: `host`, `port`  (Default: `127.0.0.1:8332`)
```
Comment (cdecker) [inline contrib/linearize/README.md]: Minor nit: the default suggests that the `ip:port` format should be used, while it really is two different options.

File: contrib/linearize/README.md [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
-* "netmagic": network magic number
-* "max_out_sz": maximum output file size (default `1000*1000*1000`)
-* "split_timestamp": Split files when a new month is first seen, in addition to
-reaching a maximum file size.
-* "file_timestamp": Set each file's last-modified time to that of the
-most recent block in that file.
+* `file_timestamp`: Set each file's last-modified time to that of the most
+recent block in that file.
+* `genesis`: The hash of the genesis block in the blockchain.
+* `input: bitcoind blocks/ directory containing blkNNNNN.dat
```
Comment (cdecker) [inline contrib/linearize/README.md]: Missing closing quote on for `input`.

File: contrib/linearize/linearize-data.py [hunk 3]
```diff
@@ -8,23 +8,33 @@
...
 from collections import namedtuple
+from binascii import hexlify, unhexlify
 
 settings = {}
 
+##### Switch endian-ness #####
+def hex_switchEndian(s):
+	""" Switches the endianness of a hex string (in pairs of hex chars) """
+	pairList = [s[i:i+2].encode() for i in range(0, len(s), 2)]
+	return b''.join(pairList[::-1]).decode()
```
Comment (cdecker) [inline contrib/linearize/linearize-data.py]: The `encode` and `decode` call seem to be unnecessary. Also my highly unscientific test shows that 
```
s.decode('hex')[::-1].encode('hex')
```
is about 5 times faster than the above loop, but may be a bit more confusing.

Comment (cdecker) [inline contrib/linearize/linearize-data.py]: `hexlify(unhexlify(s)[::-1])` is even quicker :wink:

Comment (droark) [inline contrib/linearize/linearize-data.py]: Thanks! Will fix everything up.

Comment (droark) [inline contrib/linearize/linearize-data.py]: Hello. After playing around with *timeit()*, it looks like the original code and *hexlify(unhexlify(s)[::-1]).decode()* are about the same speed. (*s.decode('hex')[::-1].encode('hex')* doesn't work, and I think it'd return incorrect results anyway.) Can you let me know how you benchmarked everything? As is, I think I'm going to leave this alone.

As for encode/decode, that has more to do with writing code that can run on Python 2 & 3; I'm also trying to maintain backwards compatibility. *hexlify(unhexlify(s)[::-1])* works by itself but the result is *b'0e0ac90a'*. That won't work without decoding it.

**Comment by droark** [comment]
Fixup based on @cdecker feedback.

File: contrib/linearize/linearize-data.py (hunk 3 above)
Comment (cdecker) [inline contrib/linearize/linearize-data.py]: Yeah, I noticed too late that the `encode`/`decode` hex variant will not work on Python3. We're probably micro-optimizing here anyway, given that the rest of the code is doing RPC calls :-)

**Comment by laanwj** [comment]
utACK 3c8f63b

**Pull request merged**